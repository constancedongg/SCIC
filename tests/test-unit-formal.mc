|'{mm} = 1000.0 '{m}|;

void func test (float '{m} start1, float '{m} end1, float '{m} start2, float '{m} end2) {
    printf(start1); /* 3 */
    printf(end1); /* 0.4 */
    printf(start2);  /* 3.01 */
    printf(end2);   /* 0.41 */
} 

int func main() {
    float[] '{mm} s1 = [105.2, 98.4, 100.3];
    float[] '{mm} e1 = [305.8, 309.9, 398.5];
    float[] '{mm} s2 = [105.1, 94.7, 101.1];
    float[] '{mm} e2 = [305.3, 310.1, 399.8];
    
    float '{mm} tmp = 3000.0;
    float '{mm} tmp2 = 400.0;
    test(s1[0], s1[1], s1[2] + 10.0, tmp2 + 10.0); 
    return 0;
}
