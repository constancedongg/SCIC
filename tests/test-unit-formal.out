0.1052
0.0984
0.1103
0.41
